tests/kafkatest/services/kafka/kafka.py

```diff
@@ -56,24 +72,34 @@ def start(self):
                 self.create_topic(topic_cfg)
 
     def start_node(self, node):
-        props_file = self.render('kafka.properties', node=node, broker_id=self.idx(node))
+        cfg = KafkaConfig(**node.config)
+        cfg[kafka_prop.ADVERTISED_HOSTNAME] = node.account.hostname
+        cfg[kafka_prop.ZOOKEEPER_CONNECT] = self.zk.connect_setting()
+        props_file = cfg.render()
         self.logger.info("kafka.properties:")
         self.logger.info(props_file)
         node.account.create_file("/mnt/kafka.properties", props_file)
 
-        cmd = "/opt/kafka/bin/kafka-server-start.sh /mnt/kafka.properties 1>> /mnt/kafka.log 2>> /mnt/kafka.log & echo $! > /mnt/kafka.pid"
+        cmd = "export LOG_DIR=/mnt/kafka-operational-logs/; "
+        cmd += self._kafka_dir(node) + "/bin/kafka-server-start.sh /mnt/kafka.properties 1>> /mnt/kafka.log 2>> /mnt/kafka.log &"
         self.logger.debug("Attempting to start KafkaService on %s with command: %s" % (str(node.account), cmd))
         with node.account.monitor_log("/mnt/kafka.log") as monitor:
             node.account.ssh(cmd)
             monitor.wait_until("Kafka Server.*started", timeout_sec=30, err_msg="Kafka server didn't finish startup")
-        if len(self.pids(node)) == 0:
-            raise Exception("No process ids recorded on node %s" % str(node))
+
+    def _kafka_dir(self, node):
+        if node.version == KafkaVersion.TRUNK:
+            return "/opt/kafka"
+        else:
+            return "/opt/kafka-" + node.version
 
     def pids(self, node):
         """Return process ids associated with running processes on the given node."""
         try:
-            return [pid for pid in node.account.ssh_capture("cat /mnt/kafka.pid", callback=int)]
-        except:
+            cmd = "ps ax | grep -i kafka.properties | grep java | grep -v grep | awk '{print $1}'"
```

Comment:
Why did the pid approach get reverted? Grepping `ps` output is extremely brittle -- it breaks pretty easily with any long commands so I'd like to avoid using it if possible.